new-integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CleanupUtil.java

```diff
@@ -0,0 +1,524 @@
+        logger.warning("Failed to list deployments");
+      }
+
+      // check if services exist
+      try {
+        if (!Kubernetes.listServices(namespace).getItems().isEmpty()) {
+          logger.info("Services still exists!!!");
+          List<V1Service> items = Kubernetes.listServices(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list services");
+      }
+
+      // check if service accounts exist
+      try {
+        if (!Kubernetes.listServiceAccounts(namespace).getItems().isEmpty()) {
+          logger.info("Service Accounts still exists!!!");
+          List<V1ServiceAccount> items = Kubernetes.listServiceAccounts(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list service accounts");
+      }
+
+      // check if ingress exist
+      try {
+        if (!Kubernetes.listIngressExtensions(namespace).getItems().isEmpty()) {
+          logger.info("Ingress Extensions still exists!!!");
+          List<ExtensionsV1beta1Ingress> items = Kubernetes.listIngressExtensions(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list Ingress Extensions");
+      }
+
+      // check if namespaced roles exist
+      try {
+        if (!Kubernetes.listNamespacedRole(namespace).getItems().isEmpty()) {
+          logger.info("Namespaced roles still exists!!!");
+          List<V1Role> items = Kubernetes.listNamespacedRole(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list namespaced roles");
+      }
+
+      // check if namespaced role bindings exist
+      try {
+        if (!Kubernetes.listNamespacedRoleBinding(namespace).getItems().isEmpty()) {
+          logger.info("Namespaced role bindings still exists!!!");
+          List<V1RoleBinding> items = Kubernetes.listNamespacedRoleBinding(namespace).getItems();
+          for (var item : items) {
+            logger.info(item.getMetadata().getName());
+          }
+          doesnotExist = false;
+        }
+      } catch (Exception ex) {
+        logger.warning(ex.getMessage());
+        logger.warning("Failed to list namespaced role bindings");
+      }
+
+      // get namespaces
```

Comment:
this won't work - the fact that you are looking through the namespace means it must exist

Reply:
The intention of the method is to check if all the artifacts in the namespace was deleted and namespace itself deleted.
When this check is made , the delete on the namespace was already called. So it is working to tell whether the namespace exists or not.

Since you asked to rename this method to nothingFoundInNamespace I cannot have this check for namespace here.
Moved the namespace and namespace delete to separate methods.